### PR TITLE
Use open_in_bin instead of open_in for Windows compatibility

### PR DESCRIPTION
**Pull request by MisterDA**

### CODE REVIEW

File: unix/file_store.ml [hunk 1]
```diff
@@ -35,7 +35,7 @@ let remove t ~digest =
 let load t ~digest =
   let path = path_of_digest t digest in
   if Sys.file_exists path then (
-    let ch = open_in path in
+    let ch = open_in_bin path in
```
Comment (talex5) [inline unix/file_store.ml:38]: Presumably the `open_out` above should be `open_out_bin` too?

File: unix/capnp_rpc_unix.ml [hunk 2]
```diff
@@ -30,7 +30,7 @@ let parse_uri s =
 module Cap_file = struct
   let load_uri path =
     try
-      let ch = open_in path in
+      let ch = open_in_bin path in
```
Comment (talex5) [inline unix/capnp_rpc_unix.ml:33]: This seems reasonable, but I'm surprised it's needed. Depending on the platform, we should end up with either `foo\n` or `foo\r\n`, and either way the following `String.trim` should sort it out. What goes wrong on Windows?

**Comment by MisterDA** [comment]
The problem is with the snippet:

```ocaml
let read_whole_file path =
    let ch = open_in path in
    let len = in_channel_length ch in
    let s = really_input_string ch len in
    close_in ch;
    s
```

So on Windows with crlf line endings, `open_in` (text mode) does the
conversion but `in_channel_length` does not and counts two characters
per line, as in binary mode; but `really_input_string` is in text mode
and sees only one character per endline. So in the end it will have
seen less characters than expected (the `len` argument) and raise
`End_of_file`.

For the URL change, I fixed it by using dos2unix on the file (that’s
where I encountered the exception). For the other change, I’ve not had
any problem but my understanding is that you’re serializing something
and my gut instinct tells me to open that in binary mode :)

I’ve also seen code such as below, and I’m wondering if it’ll cause
problems later; I’ll find a way to check that.

```ocaml
let outfile path data =
  let ch = open_out path in (* but was it text mode or bin? *)
  let data = data ^ "\n" in (* check the endline style accordingly *)
  output_string ch data;
  close_out ch
```

**Comment by talex5** [comment]
Ah, I see! I've pushed some more commits to use binary mode for saving too, and to fix the compiler warning on 4.12.

**Pull request merged**